packages/framework/fluid-static/src/utils.ts

```diff
@@ -31,35 +33,36 @@ export const isSharedObjectClass = (obj: any): obj is SharedObjectClass<any> =>
  * of both SharedObject or DataObject. This function seperates the two and returns a registery
  * of DataObject types and an array of SharedObjects.
  */
-export const parseDataObjectsFromSharedObjects = (schema: ContainerSchema):
-    [NamedFluidDataStoreRegistryEntry[], IChannelFactory[]] => {
+export const parseDataObjectsFromSharedObjects = (
+    schema: ContainerSchema,
+): [NamedFluidDataStoreRegistryEntry[], IChannelFactory[]] => {
     const registryEntries: Set<NamedFluidDataStoreRegistryEntry> = new Set();
     const sharedObjects: Set<IChannelFactory> = new Set();
 
     const tryAddObject = (obj: LoadableObjectClass<any>) => {
         if (isSharedObjectClass(obj)) {
             sharedObjects.add(obj.getFactory());
         } else if (isDataObjectClass(obj)) {
-            registryEntries.add([obj.factory.type, Promise.resolve(obj.factory)]);
+            registryEntries.add([
+                obj.factory.type,
+                Promise.resolve(obj.factory),
+            ]);
         } else {
             throw new Error(`Entry is neither a DataObject or a SharedObject`);
         }
     };
 
     // Add the object types that will be initialized
-    Object.values(schema.initialObjects).forEach((obj) => {
-        tryAddObject(obj);
-    });
-
-    // If there are dynamic object types we will add them now
-    if (schema.dynamicObjectTypes) {
-        for (const obj of schema.dynamicObjectTypes) {
-            tryAddObject(obj);
-        }
-    }
+    const dedupedObjects = new Set([
+        ...Object.values(schema.initialObjects),
+        ...(schema.dynamicObjectTypes ?? []),
+    ]);
+    dedupedObjects.forEach(tryAddObject);
 
     if (registryEntries.size === 0 && sharedObjects.size === 0) {
-        throw new Error("Container cannot be initialized without any DataTypes");
+        throw new Error(
+            "Container cannot be initialized without any DataTypes",
+        );
```

Comment:
Typically we only do it if the line would be too long, but we aren't particularly strict about preferring single-line.  If there's a linter rule that would catch these then we could consider it, but it's probably going to take some automation if we want consistency (since it's not super practical for code reviewers to count characters).